dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/VisitorLitmusPTX.java

```diff
@@ -0,0 +1,310 @@
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        String mo = ctx.sem().content;
+        String scope;
+        switch (mo) {
+            case Tag.PTX.WEAK:
+                if (ctx.scope() != null) {
+                    throw new ParsingException("Weak store instruction doesn't need scope: " + ctx.scope().content);
+                }
+                scope = Tag.PTX.SYS;
+                break;
+            case Tag.PTX.REL:
+            case Tag.PTX.RLX:
+                scope = ctx.scope().content;
+                break;
+            default:
+                throw new ParsingException("Store instruction doesn't support sem: " + mo);
+        }
+        Store store = EventFactory.PTX.newTaggedStore(object, register, mo, scope);
+        store.addFilters(ctx.store().storeProxy);
+        return programBuilder.addChild(mainThread, store);
+    }
+
+    @Override
+    public Object visitLocalConstant(LitmusPTXParser.LocalConstantContext ctx) {
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
+        return programBuilder.addChild(mainThread, EventFactory.newLocal(register, constant));
+    }
+
+    @Override
+    public Object visitLoadLocation(LitmusPTXParser.LoadLocationContext ctx) {
+        Register register = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        MemoryObject location = programBuilder.getOrNewObject(ctx.location().getText());
+        String mo = ctx.sem().content;
+        String scope;
+        switch (mo) {
+            case Tag.PTX.WEAK:
+                if (ctx.scope() != null) {
+                    throw new ParsingException("Weak load instruction doesn't need scope: " + ctx.scope().content);
+                }
+                scope = Tag.PTX.SYS;
+                break;
+            case Tag.PTX.ACQ:
+            case Tag.PTX.RLX:
+                scope = ctx.scope().content;
+                break;
+            default:
+                throw new ParsingException("Load instruction doesn't support sem: " + mo);
+        }
+        Load load = EventFactory.PTX.newTaggedLoad(register, location, mo, scope);
+        load.addFilters(ctx.load().loadProxy);
+        return programBuilder.addChild(mainThread, load);
+    }
+
+    @Override
+    public Object visitAtomConstant(LitmusPTXParser.AtomConstantContext ctx) {
+        Register register_destination = programBuilder.getOrCreateRegister(mainThread, ctx.register().getText(), getArchPrecision());
+        MemoryObject object = programBuilder.getOrNewObject(ctx.location().getText());
+        IValue constant = new IValue(new BigInteger(ctx.constant().getText()), getArchPrecision());
```

Comment:
this can probably be repalced by `(IConst) ctx.constant().accept(this)`

Reply:
Replaced